R/process_lookup_sc_demographics.R

```diff
@@ -158,106 +200,128 @@ process_lookup_sc_demographics <- function(
-        ((
-          is.na(.data$chi_postcode) |
-            !(.data$valid_pc_chi)
-        ) &
-          !(is.na(
-            .data$submitted_postcode
-          )) & .data$valid_pc_submitted) ~ .data$submitted_postcode,
-        (is.na(.data$submitted_postcode) &
-          !.data$valid_pc_submitted) ~ .data$chi_postcode
+        !is.na(submitted_postcode) ~ submitted_postcode,
+        !is.na(chi_postcode) ~ chi_postcode,
+        .default = NA
       ),
       postcode_ch_as_home = dplyr::case_when(
         !is.na(.data$submitted_postcode_ch) ~ .data$submitted_postcode_ch,
-        (is.na(.data$submitted_postcode_ch) &
-          !is.na(.data$chi_postcode_ch)) ~ .data$chi_postcode_ch,
-        (is.na(.data$submitted_postcode_ch) &
-          !is.na(.data$chi_postcode_ch)) ~ NA
+        !is.na(.data$chi_postcode_ch) ~ .data$chi_postcode_ch,
+        .default = NA
       ),
-      postcode_ch_as_home = dplyr::if_else(is.na(.data$postcode),
+      postcode_type = dplyr::case_when(
+        !is.na(submitted_postcode) ~ "submitted",
+        !is.na(chi_postcode) ~ "chi",
+        .default = "missing"
+      ),
+      postcode = dplyr::if_else(
+        is.na(.data$postcode) & !is.na(.data$postcode_ch_as_home),
         .data$postcode_ch_as_home,
-        NA
+        .data$postcode,
+        .data$postcode
       )
     ) %>%
-    dplyr::mutate(postcode_type = dplyr::case_when(
-      (.data$postcode == .data$chi_postcode) ~ "chi",
-      (.data$postcode == .data$submitted_postcode) ~ "submitted",
-      (
-        is.na(.data$submitted_postcode) &
-          is.na(.data$chi_postcode) | is.na(.data$postcode)
-      ) ~ "missing"
-    ))
-
-  sc_demog <- sc_demog %>%
-    dplyr::mutate(postcode = dplyr::if_else(
-      is.na(.data$postcode) & !is.na(.data$postcode_ch_as_home),
-      .data$postcode_ch_as_home,
-      .data$postcode,
-      .data$postcode
-    ))
-
-  # Check where the postcodes are coming from
-  sc_demog %>%
-    dplyr::count(.data$postcode_type)
-
-  # count number of replaced postcode - compare with count above
-  na_replaced_postcodes <- sc_demog %>%
-    dplyr::count(dplyr::across(tidyselect::ends_with("_postcode"), ~ is.na(.x)))
+    # arrange before missing data is filled in
+    dplyr::arrange(
+      .data$sending_location,
+      .data$social_care_id,
+      .data$financial_year,
+      .data$financial_quarter,
+      .data$extract_date
+    ) %>%
+    # add consistent_quality to indicate how long one social_care_id has been used
+    # quality: The higher, the better.
+    # This is to tackle the situation where
+    # for one CHI, two different social_care_id submitted at the same latest date.
+    dplyr::group_by(
+      .data$sending_location,
+      .data$social_care_id,
+      .data$anon_chi
+    ) %>%
+    dplyr::mutate(consistent_quality = dplyr::n_distinct(.data$period)) %>%
+    dplyr::ungroup() %>%
+    dplyr::group_by(
+      .data$sending_location,
+      .data$social_care_id
+    ) %>%
```

Comment:
```suggestion
    dplyr::ungroup() 
    
    latest_record_flag <- sc_demog_ch %>%
       # arrange before missing data is filled in
    dplyr::arrange(
      sending_location, social_care_id, financial_year,
      financial_quarter, extract_date
    ) %>%
    dplyr::group_by(
      .data$sending_location,
      .data$social_care_id
    ) %>%
```
can we please make a new dataframe here as we will incorporate the latest record flag methodology from the social care team